plat/common/include/uk/plat/common/memory.h

```diff
@@ -112,6 +112,113 @@ ukplat_memregion_list_insert(struct ukplat_memregion_list *list,
 	return (int)i;
 }
 
+/**
+ * Inserts a region into the memory region list. See
+ * ukplat_memregion_list_insert() for a detailed description. Compared to the
+ * that function, this one ignores empty regions.
+ * @param list The memory region list to insert the range into
+ * @param mrd The memory range to insert.
+ * @return Zero if the region was added successfully, a negative errno otherwise
+ */
+static inline int
+ukplat_memregion_list_insert_nonempty(struct ukplat_memregion_list *list,
```

Comment:
I don't see the need for an additional function just for this. You are unlikely to use this outside of the boot code, so why not just do an `if` statement there outside the insertion function? Maybe I am missing other use cases?

Reply:
All the functions in the header are only used in the boot code ^^
I assume you meant that the behavior doesn't make any difference in the current state of the PR. Indeed, with the last update to the PR the primary use-case was refactored away (I would say it's safe to assume that the firmware doesn't give us zero-length regions).